### DISCUSSION TITLE
How to set time of day in DateRangePicker to specific defaults?

### DISCUSSION THREAD

**Discussion by alimovz** [rsuite/rsuite#2445] [General]
Is there a way to default time of day to `00:00:00` for start date and to `23:59:59` for end date in DateRangePicker? Currently, it seems that whatever day I pick, the time of day defaults to my local computer's time, which, frankly, is rarely what's needed in a date range picker.

Predefined ranges (like Today or Yesterday) work as expected with time of day being `00:00:00` and `23:59:59`, but anything custom puts the time at whatever the time is at the moment.

**Comment by simonguo** [comment]
Hi @alimovz ,You can customize the time in the `onChange` callback.
```tsx
import parse from "date-fns/parse";

const App = () => {
  const [date, setDate] = useState();

  const handleChange = (value) => {
    if (value) {
      const start = parse("00:00:00", "HH:mm:ss", value[0]);
      const end = parse("23:59:59", "HH:mm:ss", value[1]);
      setDate([start, end]);
      return;
    }

    setDate(value);
  };

  return <DateRangePicker value={date} onChange={handleChange} />;
};
```
